File: Server/src/main/java/com/inther/dto/PresentationDto.java
package com.inther.dto;

import com.inther.assets.validators.RequestDataValidator;
import lombok.Data;

import javax.validation.constraints.*;
import java.io.Serializable;
import java.util.List;

@Data
public class PresentationDto implements Serializable
{
    @NotBlank(groups = {RequestDataValidator.UpdatePresentation.class})
    private String id;

    @NotBlank(groups = {RequestDataValidator.UpdatePresentation.class, RequestDataValidator.AddPresentation.class})
    private String email;

    @NotBlank
    private String title;

    @NotBlank
    private String description;

    @NotBlank
    private String startTime;

    @NotBlank
    private String endTime;

    @NotBlank
    private String date;

    @NotBlank
    private String place;

    private Double avgMark;
}

File: Client/src/app/models/dtos/presentation.dto.ts
import { MessageDTO } from './message.dto';
import { MarkDTO } from './mark.dto';
import { Presentation } from '../presentation.model';

export class PresentationDTO {

  public id: string;
  public email: string;
  public title: string;
  public description: string;
  public startTime: string;
  public endTime: string;
  public date: string;
  public place: string;
  public avgMark: number;

  constructor(obj: Object = {}) {
    Object.assign(this, obj);
  }

  static toModel(p: PresentationDTO): Presentation {
    return <Presentation>{
      id: p.id,
      email: p.email,
      title: p.title,
      description: p.description,
      startTime: p.startTime,
      endTime: p.endTime,
      place: p.place,
      date: p.date,
      avgMark: p.avgMark
    };
  }

}
